src/test/java/net/masterthought/cucumber/json/EmbeddingWithNameTest.java

```diff
@@ -0,0 +1,126 @@
+
+/**
+ * @author Damian Szczepanik (damianszczepanik@github)
+ * @author <a href="https://github.com/ghostcity">Stefan Gasterstädt</a>
+ */
+@RunWith(Parameterized.class)
+public class EmbeddingWithNameTest {
+
+    private static final String NO_DECODING = null;
+
+    @Parameters(name = "\"{0}\" into \"{2}\"")
+    public static Iterable<Object[]> data() {
+        return asList(new Object[][] {
+            { "application/javascript", "alert('Hello World!');", "hello-world.js", NO_DECODING, ".js" },
+            { "application/xhtml+xml; charset=UTF-8", "c29tZSBkYXRh", "data.xhtm", "some data", "embedding_-1003041823.xhtm" },
+            { "application/xslt+xml", "<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\" />", "filename-with-different-extension.xsl", NO_DECODING, ".xsl" },
+            { "application/xslt+xml", "<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\" />", "filename-without-extension", NO_DECODING, ".xslt" },
+            { "application/xslt+xml", "<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\" />", "filename-with.invalid_extension", NO_DECODING, ".xslt" },
+            { "text/jscript", "alert('Hello World!');", "hello-world.js", NO_DECODING, ".js" },
+            { "text/markdown", "c29tZSBkYXRh", "memo.md", "some data", "embedding_-1003041823.md" },
+            { "text/php", "echo 'Hello World!';", "hello-world.php7", NO_DECODING, ".php7" },
+        });
+    }
+
+    @Parameter(0)
+    public String mimeType;
+
+    @Parameter(1)
+    public String data;
+
+    @Parameter(2)
+    public String name;
+
+    @Parameter(3)
+    public String decodedData;
+
+    @Parameter(4)
+    public String fileName;
+
+    @Test
+    public void getMimeType_ReturnsMimeType() {
+        // given
+        Embedding embedding = new Embedding(this.mimeType, this.data, this.name);
+
+        // when
+        String actualMimeType = embedding.getMimeType();
+
+        // then
+        assertThat(actualMimeType).isEqualTo(this.mimeType);
+    }
+
+    @Test
+    public void getData_ReturnsContent() {
+        // given
+        Embedding embedding = new Embedding(this.mimeType, this.data, this.name);
+
+        // when
+        String actualContent = embedding.getData();
+
+        // then
+        assertThat(actualContent).isEqualTo(data);
+    }
+
+    @Test
+    public void getDecodedData_ReturnsDecodedContent() {
+        assumeThat(this.decodedData).isNotEqualTo(NO_DECODING);
+        
+        // given
+        Embedding embedding = new Embedding(this.mimeType, this.data, this.name);
+
+        // when
+        String actualDecodedContent = embedding.getDecodedData();
+
+        // then
+        assertThat(actualDecodedContent).isEqualTo(this.decodedData);
+    }
+
+    @Test
+    public void getFileName_ReturnsFileName() {
+        assumeThat(this.fileName).matches("^[^\\.]+\\.[^\\.]+$");
```

Comment:
@ghostcity I found that this assertion does not meet  "*.js" from above test and build log https://api.travis-ci.org/v3/job/753606020/log.txt says
"[WARNING] Tests run: 180, Failures: 0, Errors: 0, Skipped: 15, Time elapsed: 0.543 s - in net.masterthought.cucumber.json.EmbeddingTest"

because unknown reason - will you fix it or should I remove this code?